vzstorage-pd/virtuozzo-provisioner.go

```diff
@@ -319,18 +331,15 @@ func (p *vzFSProvisioner) Delete(volume *v1.PersistentVolume) error {
 		return err
 	}
 
-	ploopPath := path.Join(mount, options["volumePath"], options["volumeID"])
-	vol, err := ploop.PloopVolumeOpen(ploopPath)
-	if err != nil {
-		return err
-	}
-	glog.Infof("Delete: %s", ploopPath)
-	if err = vol.Delete(); err != nil {
+	if err = removePloop(mount, options); err != nil {
 		return err
 	}
 
 	newSecret := *secret
-	finalizer := fmt.Sprintf("virtuozzo.com/%s-pv", volume.UID)
+	finalizer, ok := options["finalizer"]
+	if !ok {
+		return fmt.Errorf("Unable to find finalizer in flexvolume %s options", volume.Name)
```

Comment:
I am not sure the lack of finalizer could block PV deletion.

Reply:
Ok, I will make it just a log error message